link-grammar/linkage/linkage.c

```diff
@@ -875,6 +875,38 @@ const char * linkage_get_word(const Linkage linkage, WordIdx w)
 	return linkage->word[w];
 }
 
+// Need to free the returned value of this function later:
+// the memory pointed to by bare_word
+char * linkage_get_bare_word(const Linkage linkage, WordIdx w)
+{
+	if (!linkage) return NULL;
+	if (linkage->num_words <= w) return NULL; /* bounds-check */
+
+	// first look for unknown word tag
+	char delimiter = '[';
```

Comment:
the square bracket is also used for spell guesing and regexes, this will not work.